jobs/buildhub/inventory_to_records.py

```diff
@@ -485,6 +468,10 @@ def deduplicate_entries(entries):
                     logger.exception(e)
                     continue
 
+                # Scan the list of candidates metadata (no-op if
+                # already initialized).
+                await scan_candidates(session, product)
```

Comment:
I just moved these lines to be AFTER the repeatedly called `is_build_url()`. It's a "no-op if already initialized" but it's still an `await` which means it gives the loop a chance to run other things.